File: AppSets-Android/share/src/main/java/xcj/app/share/http/model/ContentInfoList.kt
package xcj.app.share.http.model

data class ContentInfoList(
    val uri: String,
    val count: Int,
    val infoList: List<ContentInfo>
) {
    fun decode(): ContentInfoList {
        infoList.forEach { encodeInfo ->
            encodeInfo.decode()
        }
        return this
    }

    fun encode(): ContentInfoList {
        infoList.forEach { encodeInfo ->
            encodeInfo.encode()
        }
        return this
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/share/http/model/ContentInfoList.swift
//
//  ContentInfoListWrapper.swift
//  AppSets
//
//  Created by caiju Xu on 2025/3/3.
//
import Foundation

class ContentInfoList: Codable {
    var uri: String
    var count: Int
    var infoList: [ContentInfo]
    
    init(uri: String, count: Int, infoList: [ContentInfo]) {
        self.uri = uri
        self.count = count
        self.infoList = infoList
    }
    
    enum CodingKeys: CodingKey {
        case uri
        case count
        case infoList
    }
  
    func decode() -> ContentInfoList {
        infoList.forEach { info in
            _ = info.decode()
        }
        return self
    }

    func encode() -> ContentInfoList {
        infoList.forEach { info in
            _ = info.encode()
        }
        return self
    }
    
}
